nemo/collections/asr/models/classification_models.py

```diff
@@ -471,113 +652,149 @@ def change_labels(self, new_labels: List[str]):
+        return {'val_loss': loss_value, 'val_mse': val_mse, 'val_mae': val_mae}
+
+    def test_step(self, batch, batch_idx, dataloader_idx: int = 0):
+        logs = self.validation_step(batch, batch_idx, dataloader_idx)
+
+        return {'test_loss': logs['val_loss'], 'test_mse': logs['test_mse'], 'test_mae': logs['val_mae']}
+
+    def multi_validation_epoch_end(self, outputs, dataloader_idx: int = 0):
+        val_loss_mean = torch.stack([x['val_loss'] for x in outputs]).mean()
+        val_mse = self._mse.compute()
+        val_mae = self._mae.compute()
+
+        tensorboard_logs = {'val_loss': val_loss_mean, 'val_mse': val_mse, 'val_mae': val_mae}
+
+        return {'val_loss': val_loss_mean, 'val_mse': val_mse, 'val_mae': val_mae, 'log': tensorboard_logs}
+
+    def multi_test_epoch_end(self, outputs, dataloader_idx: int = 0):
+        test_loss_mean = torch.stack([x['test_loss'] for x in outputs]).mean()
+        test_mse = self._mse.compute()
+        test_mae = self._mae.compute()
+
+        tensorboard_logs = {'test_loss': test_loss_mean, 'test_mse': test_mse, 'test_mae': test_mae}
+
+        return {'test_loss': test_loss_mean, 'test_mse': test_mse, 'test_mae': test_mae, 'log': tensorboard_logs}
+
+    @torch.no_grad()
+    def transcribe(self, paths2audio_files: List[str], batch_size: int = 4) -> List[float]:
+        """
+        Generate class labels for provided audio files. Use this method for debugging and prototyping.
 
         Args:
-            config: A python dictionary which contains the following keys:
-            paths2audio_files: (a list) of paths to audio files. The files should be relatively short fragments. \
-                Recommended length per file is between 5 and 25 seconds.
+            paths2audio_files: (a list) of paths to audio files. \
+                Recommended length per file is approximately 1 second.
             batch_size: (int) batch size to use during inference. \
                 Bigger will result in better throughput performance but would use more memory.
-            temp_dir: (str) A temporary directory where the audio manifest is temporarily
-                stored.
 
         Returns:
-            A pytorch DataLoader for the given audio file(s).
-        """
-        dl_config = {
-            'manifest_filepath': os.path.join(config['temp_dir'], 'manifest.json'),
-            'sample_rate': self.preprocessor._sample_rate,
-            'labels': self.cfg.labels,
-            'batch_size': min(config['batch_size'], len(config['paths2audio_files'])),
-            'trim_silence': False,
-            'shuffle': False,
-        }
 
-        temporary_datalayer = self._setup_dataloader_from_config(config=DictConfig(dl_config))
-        return temporary_datalayer
+            A list of predictions in the same order as paths2audio_files
+        """
+        predictions = super().transcribe(paths2audio_files, batch_size, logprobs=True)
+        return [float(pred) for pred in predictions]
```

Comment:
Simply return predictions - there might be further processing done on it as a torch.tensor

Reply:
the problem in that case is with the type annotations as it's expecting to return `List[Float]`, should it return `List[str]`, well is not the case, but functionality is not affected...

Reply:
Hmm agreed, parallel to List[str] for classificaiton, lets keep this as List[float] then.